src/main/java/cat/udl/eps/softarch/mytournamentx/domain/TeamInvitation.java

```diff
@@ -7,38 +7,26 @@
 import javax.validation.constraints.NotBlank;
 import java.util.Date;
 
-@Entity @IdClass(TeamInvitationId.class)
+@Entity
 @Data
 @EqualsAndHashCode(callSuper = true)
-public class TeamInvitation extends UriEntity<TeamInvitationId> {
+public class TeamInvitation extends UriEntity<String> {
 
     @Id
-    @NotBlank
-    private String teamId;
-
-    @Id
-    @NotBlank
-    private String userId;
+    private String id;
 
     private String message;
 
-    private TeamInvitationId id;
-
     private Boolean accepted;
 
     private Date creationDate;
+    public TeamInvitation(){}
+    public TeamInvitation(String userId, String teamId,  String message){
 
-    public TeamInvitation(String teamId, String userId, String message){
-        this.teamId = teamId;
-        this.userId = userId;
-        this.id = new TeamInvitationId(teamId, userId);
+        this.id = userId + "_" + teamId;
```

Comment:
From the API it is not possible to call this constructor and you won't be able to pass teamId and userId.